### ISSUE TITLE
TitleBlock hamburger button is not accessible for keyboard or screen reader users

### ISSUE THREAD

**Issue by dougmacknz** [cultureamp/kaizen-design-system#1928]
TLDR: The hamburger button that shows on smaller viewports is marked up as a `div` but should be a `button`.

## Audit info

Type:WCAG 2
Principle: Perceivable, Operable, Robust
Guideline:1.1 Text Alternatives, 2.1 Keyboard Accessible, 4.1 Compatible
Success Criteria: 1.1.1 Non-text Content (A), 2.1.1 Keyboard (A), 4.1.2 Name, Role, Value (A)
Level: A
Platform: Web
Severity:Critical

### What's the issue?
The menu button:
* Does not have an accessible label
* Is a custom element and has not been programmatically assigned the role of button
* Keyboard users cannot tab to or activate the button

### Why it matters
All functionality must be operable using only a keyboard. Some users can only interact with their devices using a keyboard or similar input mechanism.

If the correct role is not applied a screen reader user will not know how to interact with it.

If the button does not have a meaningful name, a screen reader user will not know what the button is for.

### Our recommendation
Option 1: Use a native `<button>` element
* Mark up the button with the `<button>` element
* This will ensure it is keyboard accessible
* Give the `<button>` an aria-label with a meaningful label

Code example
```
<button aria-label="menu" class="TitleBlockZen-TitleBlockZen_hamburger--3NuBf">
  <svg class="Icon-Icon-module_icon--20XXx" viewBox="0 0 20 20" focusable="false" role="presentation" aria-hidden="true">...</svg>
</button>
```

Option 2: add keyboard support to existing markup
* Add `role="button"` to the `<div>`
* Add `tabindex="0"` to the `<div>`
* Give the `<div>` an aria-label with a meaningful label

Code example
```
<div role="button" aria-label="menu" tabindex="0" class="TitleBlockZen-TitleBlockZen_hamburger--3NuBf">
  <svg class="Icon-Icon-module_icon--20XXx" viewBox="0 0 20 20" focusable="false" role="presentation" aria-hidden="true">...</svg>
</div>
```
